### DISCUSSION TITLE
Installation steps

### DISCUSSION THREAD

**Discussion by lvaoliveira** [serkri/SmartEVSE-3#54] [Q&A] [answered]
Hey there,

I'm really impressed by the project and I got a SmartEvse.

I did some research and I couldn't find any concrete information about how to upload your firmware into the controller.

Could you please advise me on how to do it?

Thank you

**Comment by dingo35** [comment, marked as answer]
Goto http://your-ip/update; upload the files firmware.bin and spiffs.bin ; that's all!

You can find your-ip (the SmartEVSE ip address) on your router OR on the wifi-setup screen (small line on top of lcd screen) of the SmartEVSE.

**Reply by lvaoliveira** [reply]
Perfect, thank you!!!

**Reply by heijer** [reply]
I have some follow-up questions:
 - Does the order of uploading the files firmware.bin and spiffs.bin make a difference?
 - Do you always need to update both? If not, how can we know which of the two needs to be updated?

**Reply by dingo35** [reply]
I usually first upgrade firmware.bin, and then spiffs.bin ; temporarily running firmware.bin with an "old" spiffs.bin has never resulted in strange results, it only affects the webserver part and you only use the "update" routine to reflash spiffs.bin.

Only if you upgrade from the original firmware to the serkri distribution you will notice strange things, but the /update still works so reflashing spiffs.bin should give no problems.

**Reply by dingo35** [reply]
Spiffs.bin only needs to be updated when files in the data/ directory have changed; you can see that here:
https://github.com/serkri/SmartEVSE-3/tree/master/SmartEVSE-3/data

**Reply by dingo35** [reply]
Firmware.bin _always_ should be updated.

**Reply by heijer** [reply]
Thanks a lot for explaining!
